amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
Though, I don't actually have that use case at the moment - just inferring it from @dannf